### PR TITLE
Add check so you can only change level when icon reaches destination …

### PR DESCRIPTION
**Pull request by vchadha**
…node

Is slightly better

we could have a hardcode check for position, but i think this is good enough for now

### CODE REVIEW

File: Assets/Scripts/SceneControllers/LevelSelectController.cs [hunk 1]
```diff
@@ -57,12 +59,14 @@ protected void Start()
...
         LoadLevels();
+
+        isMovingIcon = false;
     }
 
     override protected void SceneUpdate()
     {
         // Enter level
-        if (Input.GetKeyDown(KeyCode.Return))
+        if (Input.GetKeyDown(KeyCode.Return) && !isMovingIcon)
```
Comment (justinro-underscore) [inline Assets/Scripts/SceneControllers/LevelSelectController.cs:69]: Can we add this check to the left and right arrow actions too? I think it'll feel better

File: Assets/Scripts/SceneControllers/LevelSelectController.cs [hunk 2]
```diff
@@ -105,8 +109,12 @@ private void ChangeSelectedLevel()
         // Update selected level
         selectedLevel = levels[selectedLevelIndex];
 
+        // We are moving icon
+        isMovingIcon = true;
+
         // Move player icon to new selected level icon
-        playerTransform.DOMove(selectedLevel.GetLevelIconLocation(), playerIconSpeed, false);
+        playerTransform.DOMove(selectedLevel.GetLevelIconLocation(), playerIconSpeed, false)
+            .OnComplete(() => isMovingIcon = false);
```
Comment (justinro-underscore) [inline Assets/Scripts/SceneControllers/LevelSelectController.cs:117]: Love a callback 😍

**Review by justinro-underscore — CHANGES_REQUESTED** [review]

File: Assets/Scripts/SceneControllers/LevelSelectController.cs (hunk 1 above)
Comment (vchadha) [reply Assets/Scripts/SceneControllers/LevelSelectController.cs:69]: oooo good call

**Review by justinro-underscore — APPROVED** [review]
🏃 🎽 👟

**Pull request merged**